react/components/Preview.jsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+class Preview extends React.Component {
+  constructor (props) {
+    super(props);
+    this.state = {
+      imgSource       : '',
+      defaultThumbnail: '/assets/img/video_thumb_default.png',
+    };
+    this.previewFile = this.previewFile.bind(this);
+  }
+  componentWillMount () {
+    console.log('Preview will mount');
+    this.previewFile(this.props.file);
```

Comment:
Generally you should avoid doing dom stuff/setting state in `componentWillMount`. Especially since it seems that they will be removing it in a future React version.

`componentDidMount` is preferred